src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -139,42 +146,15 @@ public async Task<object> GetDescriptionAsync(IAsyncCompletionSession session, V
             }
 
             var service = document.GetLanguageService<CompletionService>();
-            var description = await service.GetDescriptionAsync(document, roslynItem, cancellationToken).ConfigureAwait(false);
-
-            return new ClassifiedTextElement(description.TaggedParts.Select(p => new ClassifiedTextRun(p.Tag.ToClassificationTypeName(), p.Text)));
-        }
 
-        private static RoslynTrigger GetRoslynTrigger(AsyncCompletionData.CompletionTrigger trigger, SnapshotPoint triggerLocation)
-        {
-            switch (trigger.Reason)
+            if (service== null)
             {
-                case AsyncCompletionData.CompletionTriggerReason.Invoke:
-                case AsyncCompletionData.CompletionTriggerReason.InvokeAndCommitIfUnique:
-                    return RoslynTrigger.Invoke;
-                case AsyncCompletionData.CompletionTriggerReason.Insertion:
-                    return RoslynTrigger.CreateInsertionTrigger(trigger.Character);
-                case AsyncCompletionData.CompletionTriggerReason.Deletion:
-                    return RoslynTrigger.CreateDeletionTrigger(trigger.Character);
-                case AsyncCompletionData.CompletionTriggerReason.Backspace:
-                    var snapshotBeforeEdit = trigger.ViewSnapshotBeforeTrigger;
-                    char characterRemoved;
-                    if (triggerLocation.Position >= 0 && triggerLocation.Position < snapshotBeforeEdit.Length)
-                    {
-                        // If multiple characters were removed (selection), this finds the first character from the left. 
-                        characterRemoved = snapshotBeforeEdit[triggerLocation.Position];
-                    }
-                    else
-                    {
-                        characterRemoved = (char)0;
-                    }
-
-                    return RoslynTrigger.CreateDeletionTrigger(characterRemoved);
-                case AsyncCompletionData.CompletionTriggerReason.SnippetsMode:
-                    return new RoslynTrigger(CompletionTriggerKind.Snippets);
+                return null;
             }
 
-            FatalError.ReportWithoutCrash(ExceptionUtilities.UnexpectedValue(trigger.Reason));
-            return RoslynTrigger.Invoke;
+            var description = await service.GetDescriptionAsync(document, roslynItem, cancellationToken).ConfigureAwait(false);
+
+            return new ClassifiedTextElement(description.TaggedParts.Select(p => new ClassifiedTextRun(p.Tag.ToClassificationTypeName(), p.Text)));
```

Comment:
this looks like a duplicate of:

```c#
        private static ClassifiedTextElement BuildClassifiedTextElement(QuickInfoSection section)
        {
            return new ClassifiedTextElement(section.TaggedParts.Select(
                    part => new ClassifiedTextRun(part.Tag.ToClassificationTypeName(), part.Text)));
        }
```

Extract helper to just work on an `ImmutableArray<TaggedText>`?